src/convertToObject.js

```diff
@@ -3,16 +3,29 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const res = {};
+  const arr = sourceString
+    .replace(/\s{2,}|^\s/g, '')
+    .split(';')
+    .filter(a => a.length > 2);
+
+  for (let i = 0; i < arr.length; i++) {
+    arr[i] = arr[i].trim().replace(/\s(?=:)|^\s/gm, '');
```

Comment:
You can split all elements by : separator, trim them using map method and then just add final values to the object
Such way will be more understandable and help to avoid so many matches